File: CS/Helper/GridRadioGroupColumnHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraEditors;

namespace WindowsApplication1
{
    public class GridRadioGroupColumnHelper
    {

        public event EventHandler SelectedRowChanged;
        private GridView _GridView;

        private RepositoryItemCheckEdit _RepositoryItem = new RepositoryItemCheckEdit();
        public RepositoryItemCheckEdit RadioRepositoryItem
        {
            get { return _RepositoryItem; }
            set { _RepositoryItem = value; }
        }

        private GridColumn _RadioGroupColumn = new GridColumn();
        public GridColumn RadioGroupColumn
        {
            get { return _RadioGroupColumn; }
            set { _RadioGroupColumn = value; }
        }


        private int _SelectedDataSourceRowIndex;
        public int SelectedDataSourceRowIndex
        {
            get { return _SelectedDataSourceRowIndex; }
            set
            {
                if (_SelectedDataSourceRowIndex != value)
                {
                    int oldRowIndex = _SelectedDataSourceRowIndex;
                    _SelectedDataSourceRowIndex = value;
                    SetRowChecked(oldRowIndex, false);
                    OnSelectedRowChanged(oldRowIndex, value);
                }
            }
        }

        private void SetRowChecked(int dataSourceRowIndex, bool value)
        {
            int rowHandle = _GridView.GetRowHandle(dataSourceRowIndex);
            _GridView.SetRowCellValue(rowHandle, RadioGroupColumn, value);
        }
        public GridRadioGroupColumnHelper(GridView gridView)
        {
            _GridView = gridView;
            _GridView.BeginUpdate();
            InitGridView();
            InitColumn();
            InitRepositoryItem();
            _GridView.EndUpdate();
        }
        private void InitGridView()
        {
            _GridView.CustomUnboundColumnData += _GridView_CustomUnboundColumnData;
        }

        void _GridView_CustomUnboundColumnData(object sender, CustomColumnDataEventArgs e)
        {
            if (e.Column == RadioGroupColumn)
            {
                if (e.IsGetData)
                    e.Value = e.ListSourceRowIndex == SelectedDataSourceRowIndex;
                if (e.IsSetData)
                    if (e.Value.Equals(true))
                        SelectedDataSourceRowIndex = e.ListSourceRowIndex;
            }
        }
        private void InitColumn()
        {
            RadioGroupColumn.FieldName = "RadioGroupColumn";
            _GridView.Columns.Add(RadioGroupColumn);
            RadioGroupColumn.Visible = true;
            RadioGroupColumn.ColumnEdit = RadioRepositoryItem;
            RadioGroupColumn.Caption = "Radio";
            RadioGroupColumn.UnboundType = DevExpress.Data.UnboundColumnType.Boolean;
            RadioGroupColumn.MaxWidth = 50;
        }
        private void InitRepositoryItem()
        {
            RadioRepositoryItem.CheckStyle = DevExpress.XtraEditors.Controls.CheckStyles.Radio;
            RadioRepositoryItem.EditValueChanged += RadioRepositoryItem_EditValueChanged;
            _GridView.GridControl.RepositoryItems.Add(RadioRepositoryItem);
        }


        void RadioRepositoryItem_EditValueChanged(object sender, EventArgs e)
        {
            _GridView.PostEditor();
        }

        private void OnSelectedRowChanged(int oldValue, int newValue)
        {
            RaiseSelectedRowChanged();
        }
        protected virtual void RaiseSelectedRowChanged()
        {
            if (SelectedRowChanged != null)
                SelectedRowChanged(_GridView, EventArgs.Empty);
        }
        public void Disable() {
            RadioRepositoryItem.EditValueChanged -= RadioRepositoryItem_EditValueChanged;
            _GridView.CustomUnboundColumnData -= _GridView_CustomUnboundColumnData;
            _GridView = null;
            RadioRepositoryItem = null;
        }
    }
}

File: VB/Helper/GridRadioGroupColumnHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraEditors

Namespace WindowsApplication1
	Public Class GridRadioGroupColumnHelper

		Public Event SelectedRowChanged As EventHandler
		Private _GridView As GridView

		Private _RepositoryItem As New RepositoryItemCheckEdit()
		Public Property RadioRepositoryItem() As RepositoryItemCheckEdit
			Get
				Return _RepositoryItem
			End Get
			Set(ByVal value As RepositoryItemCheckEdit)
				_RepositoryItem = value
			End Set
		End Property

		Private _RadioGroupColumn As New GridColumn()
		Public Property RadioGroupColumn() As GridColumn
			Get
				Return _RadioGroupColumn
			End Get
			Set(ByVal value As GridColumn)
				_RadioGroupColumn = value
			End Set
		End Property


		Private _SelectedDataSourceRowIndex As Integer
		Public Property SelectedDataSourceRowIndex() As Integer
			Get
				Return _SelectedDataSourceRowIndex
			End Get
			Set(ByVal value As Integer)
				If _SelectedDataSourceRowIndex <> value Then
					Dim oldRowIndex As Integer = _SelectedDataSourceRowIndex
					_SelectedDataSourceRowIndex = value
					SetRowChecked(oldRowIndex, False)
					OnSelectedRowChanged(oldRowIndex, value)
				End If
			End Set
		End Property

		Private Sub SetRowChecked(ByVal dataSourceRowIndex As Integer, ByVal value As Boolean)
			Dim rowHandle As Integer = _GridView.GetRowHandle(dataSourceRowIndex)
			_GridView.SetRowCellValue(rowHandle, RadioGroupColumn, value)
		End Sub
		Public Sub New(ByVal gridView As GridView)
			_GridView = gridView
			_GridView.BeginUpdate()
			InitGridView()
			InitColumn()
			InitRepositoryItem()
			_GridView.EndUpdate()
		End Sub
		Private Sub InitGridView()
			AddHandler _GridView.CustomUnboundColumnData, AddressOf _GridView_CustomUnboundColumnData
		End Sub

		Private Sub _GridView_CustomUnboundColumnData(ByVal sender As Object, ByVal e As CustomColumnDataEventArgs)
			If e.Column Is RadioGroupColumn Then
				If e.IsGetData Then
					e.Value = e.ListSourceRowIndex = SelectedDataSourceRowIndex
				End If
				If e.IsSetData Then
					If e.Value.Equals(True) Then
						SelectedDataSourceRowIndex = e.ListSourceRowIndex
					End If
				End If
			End If
		End Sub
		Private Sub InitColumn()
			RadioGroupColumn.FieldName = "RadioGroupColumn"
			_GridView.Columns.Add(RadioGroupColumn)
			RadioGroupColumn.Visible = True
			RadioGroupColumn.ColumnEdit = RadioRepositoryItem
			RadioGroupColumn.Caption = "Radio"
			RadioGroupColumn.UnboundType = DevExpress.Data.UnboundColumnType.Boolean
			RadioGroupColumn.MaxWidth = 50
		End Sub
		Private Sub InitRepositoryItem()
			RadioRepositoryItem.CheckStyle = DevExpress.XtraEditors.Controls.CheckStyles.Radio
			AddHandler RadioRepositoryItem.EditValueChanged, AddressOf RadioRepositoryItem_EditValueChanged
			_GridView.GridControl.RepositoryItems.Add(RadioRepositoryItem)
		End Sub


		Private Sub RadioRepositoryItem_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
			_GridView.PostEditor()
		End Sub

		Private Sub OnSelectedRowChanged(ByVal oldValue As Integer, ByVal newValue As Integer)
			RaiseSelectedRowChanged()
		End Sub
		Protected Overridable Sub RaiseSelectedRowChanged()
			RaiseEvent SelectedRowChanged(_GridView, EventArgs.Empty)
		End Sub

		Public Sub Disable()
			RemoveHandler RadioRepositoryItem.EditValueChanged, AddressOf RadioRepositoryItem_EditValueChanged
			RemoveHandler _GridView.CustomUnboundColumnData, AddressOf _GridView_CustomUnboundColumnData
			_GridView = Nothing
			RadioRepositoryItem = Nothing
		End Sub

	End Class
End Namespace
